packages/react/src/hooks/useAuth.ts

```diff
@@ -27,32 +23,74 @@ export const useAuth = (): UseAuthResult => {
     user: undefined,
   });
 
-  const handleAuth: HubCallback = ({ payload }) => {
-    switch (payload.event) {
-      case 'signIn':
-        return setResult({ user: payload.data as AuthUser, isLoading: false });
-      case 'signOut':
-        return setResult({ isLoading: false });
-      default:
-        break;
+  /**
+   * Hub events like `tokenRefresh` will not give back the user object.
+   * This util will be used to get current user after those events.
+   */
+  const fetchCurrentUser = React.useCallback(async () => {
+    setResult({ user: undefined, isLoading: true, error: undefined });
+
+    try {
+      // casting the result because `Auth.currentAuthenticateduser` returns `any`
+      const user = (await Auth.currentAuthenticatedUser()) as AmplifyUser;
+      setResult({ user, isLoading: false });
+    } catch (e) {
+      const error = e as Error;
```

Comment:
what's this line for?

Reply:
Ah, so I needed to cast `e` to an `error`, because `e` was `any`.

Reply:
I was hoping we could do something like
```ts
} catch (error: Error) {
  setResult({ error, isLoading: false })
}
```

but TS does [not support](https://github.com/microsoft/TypeScript/issues/20024) strongly typing `catch` statement, so left it like that.

Reply:
I see. Thanks!